src/black/__init__.py

```diff
@@ -1393,6 +1393,22 @@ def patch_click() -> None:
 
 
 def patched_main() -> None:
+    #: Fixes errors with emoji in  Windows terminals when output is redirected
+    # (i.e. pre-commit): https://github.com/psf/black/issues/3156
+    # To be as safe as possible, we only enable this for "Windows 10 October
+    # 2018 Update (version 1809)" and up
+    if (
+        "pytest" not in sys.modules
+        and platform.system() == "Windows"
+        and tuple(map(int, platform.version().split("."))) >= (10, 0, 1809)
```

Comment:
Can't decide if I love or hate that last line. Advantages:
- it's correct (as it uses `int` comparisons);
- it performs well (as the `and` short circuits for non-Windows users);
- fits in a single line.

The disadvantage is that it reads pretty clumsily ("make a tuple out of an int map of the string that `platform.version()` returns split by the period... and then compare it to (10, 0, 1809)"). Especially that int map bit.

But I guess the advantages outweigh my personal preference here.

Reply:
Pretty much sums up my feelings when writing it!